src/java.desktop/share/classes/javax/swing/ProgressMonitorInputStream.java

```diff
@@ -145,7 +151,13 @@ public int read(byte[] b,
                     int off,
                     int len) throws IOException {
         int nr = in.read(b, off, len);
-        if (nr > 0) monitor.setProgress(nread += nr);
+        if (nr > 0) {
+            if (nread + nr > nread) {
+                monitor.setProgress(nread += nr);
+            } else {
+                monitor.setProgress(monitor.getMaximum());
+            }
+        }
```

Comment:
Looks like Lines 154-167 duplicates code from lines 129-142.
Can we consider to extract it so another method?